contracts/libraries/Errors.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+error ErrInvalidThreshold(bytes4 msgSig);
+error ErrOnlySelfCall(bytes4 msgSig);
+error ErrUnauthorized(bytes4 msgSig);
```

Comment:
In general, we often need to know details of the `Unauthorized`, especially the role. Consider to breakdown this error into more detailed ones.

Reply:
updated.